### PR TITLE
Model 1 Implementation

### PR DESCRIPTION
**Pull request by patricktnast**
### Description

- *Category*:  Implementation
- *JIRA issue*: [MIC-4101](https://jira.ihme.washington.edu/browse/MIC-4101)
- *Research reference*:  [Pregnancy Model](https://vivarium-research.readthedocs.io/en/latest/models/other_models/pregnancy/gbd_2021_closed_cohort/index.html)

### Changes and notes
 
Add postpartum state with 40 week dwell time, with transitions

### Verification and Testing

Simulants move to postpartum states after 40 weeks
CI passing

### CODE REVIEW

**Comment by patricktnast** [comment]
Uh Oh, vivarium initializes simulants at a random point in their disease dwell time.

**Comment by patricktnast** [comment]
I didn't see an obvious way out of writing a DiseaseState subclass. There are a few things there that might not be 100% necessary given our modeling assumptions, but it works.

**Comment by patricktnast** [comment]
> I didn't see an obvious way out of writing a DiseaseState subclass. There are a few things there that might not be 100% necessary given our modeling assumptions, but it works.

File: src/vivarium_gates_nutrition_optimization/components/pregnancy.py [hunk 1]
```diff
@@ -1,26 +1,48 @@
...
 
 class NotPregnantState(SusceptibleState):
     def __init__(self, cause, *args, **kwargs):
         super(SusceptibleState, self).__init__(cause, *args, name_prefix="not_", **kwargs)
 
 
+class PregnantState(DiseaseState):
+    def get_initial_event_times(self, pop_data: SimulantData) -> pd.DataFrame:
+        return pd.DataFrame(
+            {self.event_time_column: self.clock(), self.event_count_column: 1},
```
Comment (stevebachmeier) [inline src/vivarium_gates_nutrition_optimization/components/pregnancy.py:16]: Should the event time column be clock time + step size? @rmudambi ?

File: src/vivarium_gates_nutrition_optimization/components/pregnancy.py [hunk 2]
```diff
@@ -1,26 +1,48 @@
...
             "excess_mortality_rate": lambda *_: 0.0,
+            "dwell_time": lambda *_: pd.Timedelta(days=40 * 7),
+        },
+    )
+    postpartum = DiseaseState(
+        models.POSTPARTUM_STATE_NAME,
+        get_data_functions={
+            "prevalence": lambda *_: 0.0,
+            "disability_weight": lambda *_: 0.0,
+            "excess_mortality_rate": lambda *_: 0.0,
```
Comment (rmudambi) [inline src/vivarium_gates_nutrition_optimization/components/pregnancy.py:37]: This state needs a dwell time as well (I think it's 6 weeks?)

File: src/vivarium_gates_nutrition_optimization/components/pregnancy.py [hunk 3]
```diff
@@ -1,26 +1,48 @@
...
         },
     )
     pregnant.allow_self_transitions()
+    pregnant.add_transition(postpartum)
+
+    postpartum.allow_self_transitions()
```
Comment (rmudambi) [inline src/vivarium_gates_nutrition_optimization/components/pregnancy.py:43]: After that dwell time, we either need to transition simulants to not-pregnant or un-track them.

**Review by rmudambi — CHANGES_REQUESTED** [review]

File: src/vivarium_gates_nutrition_optimization/components/pregnancy.py (hunk 1 above)
Comment (rmudambi) [reply src/vivarium_gates_nutrition_optimization/components/pregnancy.py:16]: I've talked with Patrick and he will test which is correct

Comment (patricktnast) [reply src/vivarium_gates_nutrition_optimization/components/pregnancy.py:16]: you're right, we only get 39 weeks of pregnant time with this.

**Comment by patricktnast** [comment]
confirmed in interactive context that simulants go from pregnant -> postpartum -> not pregnant with appropriate person times

File: src/vivarium_gates_nutrition_optimization/components/pregnancy.py (hunk 2 above)
Comment (patricktnast) [reply src/vivarium_gates_nutrition_optimization/components/pregnancy.py:45]: done

File: src/vivarium_gates_nutrition_optimization/components/pregnancy.py (hunk 3 above)
Comment (patricktnast) [reply src/vivarium_gates_nutrition_optimization/components/pregnancy.py:52]: done

**Pull request merged**